app/controllers/helpers/CheckYourAnswersHelper.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.helpers
+
+import controllers.routes
+import models.TraderGoodsProfile
+import play.api.mvc.Result
+import play.api.mvc.Results.Redirect
+import uk.gov.hmrc.govukfrontend.views.Aliases._
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.Key
+
+class CheckYourAnswersHelper {
+
+  private def createSummaryListRow(href: String, key: String, value: String): SummaryListRow = SummaryListRow(
+    key = Key(HtmlContent(key)),
+    value = Value(HtmlContent(value)),
+    actions = Some(Actions(items = Seq(ActionItem(href, HtmlContent("Change")))))
+  )
+
+  private def createOptionalSummaryListRow(href: String, key: String, value: Option[String]): Option[SummaryListRow] =
+    if (value.isEmpty) {
+      None
+    } else {
+      Some(createSummaryListRow(href, key, value.get))
+    }
+
+  private def createOptionalYesNoSummaryListRow(
+    href: String,
+    key: String,
+    value: Option[Boolean]
+  ): Option[SummaryListRow] =
+    if (value.isEmpty) {
+      None
+    } else {
+      Some(createSummaryListRow(href, key, if (value.get) "Yes" else "No"))
+    }
+
+  def createSummaryList(traderGoodsProfile: TraderGoodsProfile): List[SummaryListRow] =
+    List(
+      createOptionalSummaryListRow(
+        routes.UkimsNumberController.onPageLoad.url,
+        "UKIMS number",
+        traderGoodsProfile.ukimsNumber match {
+          case Some(x) => Some(x.value)
+          case None    => None
+        }
+      ),
+      createOptionalYesNoSummaryListRow(
+        routes.NirmsQuestionController.onPageLoad.url,
+        "NIRMS registered",
+        traderGoodsProfile.hasNirms
+      ),
+      createOptionalSummaryListRow(
+        routes.NirmsNumberController.onPageLoad.url,
+        "NIRMS number",
+        traderGoodsProfile.nirmsNumber match {
```

Comment:
A simpler way to do this: 
`traderGoodsProfile.ukimsNumber.map(_.value)`

If it's None, it will remain as None, if there's a Some of Ukims then we can extract the value in the Some and convert it to a Some[String] in the map

Reply:
done